token/program-2022/src/pod.rs

```diff
@@ -95,16 +110,69 @@ impl From<OptionalNonZeroElGamalPubkey> for Option<ElGamalPubkey> {
         }
     }
 }
+#[cfg(feature = "serde-traits")]
+impl Serialize for OptionalNonZeroElGamalPubkey {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        if self.0 == ElGamalPubkey::default() {
+            s.serialize_none()
+        } else {
+            s.serialize_some(&self.0.to_string())
+        }
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+struct OptionalNonZeroElGamalPubkeyVisitor;
+#[cfg(feature = "serde-traits")]
+impl<'de> Visitor<'de> for OptionalNonZeroElGamalPubkeyVisitor {
+    type Value = OptionalNonZeroElGamalPubkey;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an ElGamal public key as base64")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+    {
+        let bytes = BASE64_STANDARD.decode(v).map_err(de::Error::custom)?;
```

Comment:
By convention, we typically use base58 for pubkeys. Is there a reason to use base64 for these?

Reply:
I will switch to base58. 
The reason I used base64 because that's how `fmt::Display` was implemented for it in the `solana-zk-token-sdk` crate.

Reply:
Ah I see, you might be right then! Let's see what @samkim-crypto thinks, since he did the work on `solana-zk-token-sdk`

Reply:
Yes, base58 is typicaly used for public keys and should really be used for ElGamal public keys as well. But currently, base64 is used for `fmt::Display` in the `solana-zk-token-sdk`, so here, I think it would make sense to stick with base64.

Reply:
Since this is meant to be consumed by external users, maybe we should do whatever makes the most sense at the top-level, since we can easily fixup `solana-zk-token-sdk` right away.

@jordansexton from a wallet perspective, do you think ElGamal keys for encryption should use base58 or base64?

Reply:
Another `solana-zk-token-sdk` type that is used in instruction data is `AeCiphertext` and it also uses base64. Just FYI.

Reply:
We discussed this a bit and decided that base64 is ok https://discord.com/channels/428295358100013066/774014770402689065/1138612960332349532 -- all good then!